lib/private/AppConfig.php

```diff
@@ -728,12 +743,17 @@ private function setTypedValue(
 		 * no update if key is already known with set lazy status, or value is
 		 * different, or sensitivity switched from false to true.
 		 */
-		if ($this->hasKey($app, $key, $lazy)
+		if (!$sensitive
+			&& $this->hasKey($app, $key, $lazy)
 			&& $value === $this->getTypedValue($app, $key, $value, $lazy, $type)
-			&& (!$sensitive || $this->isSensitive($app, $key, $lazy))) {
+			&& !$this->isSensitive($app, $key, $lazy)) {
 			return false;
 		}
 
+		if ($sensitive) {
```

Comment:
firefox crashed and seems suggestion was not attached to my previous review ...

`$sensitive` can be false, but once set as sensitive in database it won't be reverted unless using `updateSensitive()`

```suggestion
		if ($sensitive || $this->isSensitive($app, $key, $lazy)) {
```

Reply:
Shouldn't `$sensitive = $this->isTyped(self::VALUE_SENSITIVE, $type);` above then have this?
Also we would then need to manipulate the type and make sure we store it, even when it was not provided.

All those methods are quite complex and all invoking each others and it's really not straight forward atm for me to see what is right.

Reply:
Got it now, and added it in a second commit